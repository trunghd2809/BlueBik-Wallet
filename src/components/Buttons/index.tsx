import React from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';

import Layout from '../../constants/Layout.constants';

interface ButtonProps {
  label: string;
  onPress: () => void;
}

const Button: React.FC<ButtonProps> = ({ label, onPress }) => {
  return (
    <TouchableOpacity style={styles.button} onPress={onPress}>
      <Text style={styles.buttonText}>{label}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    backgroundColor: 'blue',
    padding: Layout.spacing.md,
    borderRadius: 4,
    alignItems: 'center',
    justifyContent: 'center',
    marginVertical: Layout.spacing.md,
  },
  buttonText: {
    color: 'white',
    fontSize: Layout.fontSize.large,
    fontWeight: 'bold',
  },
});

export { Button };
