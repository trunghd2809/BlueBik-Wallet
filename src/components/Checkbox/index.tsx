import React from 'react';
import { Image, StyleSheet, TouchableOpacity } from 'react-native';

interface IProps {
  value: boolean;
  onPress: () => void;
  size?: number;
  backgroundColor?: string;
}

export const CheckBoxButton = (props: IProps) => {
  const { value, onPress, size, backgroundColor } = props;

  return (
    <TouchableOpacity
      activeOpacity={0.7}
      style={[
        styles.checkBoxContainer,
        {
          height: size || 38,
          width: size || 38,
          backgroundColor: value ? backgroundColor : 'transparent',
          borderColor: 'blue',
        },
      ]}
      onPress={onPress}
    >
      {value ? <Image style={styles.activeIcon} source={require('../../../assets/check.png')} /> : <></>}
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  activeIcon: {
    width: 32,
    height: 32,
  },
  checkBoxContainer: {
    borderRadius: 5,
    borderWidth: 1.3,
  },
});