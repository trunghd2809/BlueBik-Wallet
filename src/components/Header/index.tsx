import React from 'react';
import { View, StyleSheet, Text } from 'react-native';

import Layout from '../../constants/Layout.constants';

interface HeaderProps {
  // children: React.ReactNode;
  label: string,
}

const Header: React.FC<HeaderProps> = ({ label }) => {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>{label}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignContent: 'center',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'transparent'
  },
  title: {
    fontSize: Layout.fontSize.giant,
  }
});

export { Header };
