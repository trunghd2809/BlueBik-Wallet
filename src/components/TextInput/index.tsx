import React from 'react';
import { View, TextInput, Text, StyleSheet, TextInputProps } from 'react-native';

import Layout from '../../constants/Layout.constants';

interface BTextInputProps extends TextInputProps {
  label: string;
  borderColor?: string;
  errors?: string | null,
}

const BTextInput: React.FC<BTextInputProps> = ({
  label,
  borderColor = 'gray',
  errors,
  ...restProps
}) => {
  return (
    <View style={styles.container}>
      <Text style={styles.label}>
        {label} 
        <Text style={styles.require}>*</Text>
      </Text>
      <TextInput
        style={[styles.input, { borderColor }]}
        {...restProps}
      />
      {errors && <Text style={styles.errors}>{errors}</Text>}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginBottom: 10,
  },
  label: {
    marginBottom: 4,
    fontSize: Layout.fontSize.medium,
  },
  input: {
    borderBottomWidth: 1,
    paddingVertical: 8,
    fontSize: 16,
    borderWidth: 1,
    borderRadius: 4,
    padding: Layout.spacing.sm,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: -2 },
    shadowOpacity: 0.1,
    shadowRadius: 5,
  },
  require: {
    color: 'red',
  },
  errors: {
    color: 'red',
    fontStyle: 'italic',
    fontSize: Layout.fontSize.small,
    marginTop: 5,
  },
});

export { BTextInput };
