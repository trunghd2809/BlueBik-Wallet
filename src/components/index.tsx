export * from './TextInput';
export * from './Buttons';
export * from './Header';
export * from './Checkbox';