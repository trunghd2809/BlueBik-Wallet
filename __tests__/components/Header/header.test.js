import React from 'react';
import renderer from 'react-test-renderer';

import { Header } from '../../../src/components';

test('renders header correctly', () => {
  const onPress = jest.fn();
  const tree = renderer.create(<Header label="Information"/>).toJSON();
  expect(tree).toMatchSnapshot();
});