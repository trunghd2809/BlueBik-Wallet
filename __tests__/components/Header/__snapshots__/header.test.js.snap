// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders header correctly 1`] = `
<View
  style={
    {
      "alignContent": "center",
      "alignItems": "center",
      "backgroundColor": "transparent",
      "flex": 1,
      "fontSize": 16,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Information
  </Text>
</View>
`;
